libfwupd/fwupd-bios-attr.c

```diff
@@ -470,17 +480,63 @@ fwupd_bios_attr_update_current_value(FwupdBiosAttr *self, const gchar *value)
 gboolean
 fwupd_bios_attr_set_current_value(FwupdBiosAttr *self, const gchar *value, GError **error)
 {
-	FwupdBiosAttrPrivate *priv = GET_PRIVATE(self);
 	g_autofree gchar *path = NULL;
+	int fd;
+	ssize_t n;
+	int r;
 
 	g_return_val_if_fail(FWUPD_IS_BIOS_ATTR(self), FALSE);
+	g_return_val_if_fail(value != NULL, FALSE);
 
 	path = fwupd_bios_attr_build(self, error);
 	if (path == NULL)
 		return FALSE;
 
-	if (!g_file_set_contents(path, value, -1, error)) {
-		g_prefix_error(error, "failed to set attribute %s to %s: ", priv->name, value);
+	fd = open(path, O_WRONLY | O_CLOEXEC);
+	if (fd < 0) {
```

Comment:
why all this low level writing? Could we use FuIOChannel instead?

Reply:
Using Glib for writing it causes problems because you can't make temp files in sysfs.  FUIOChannel might work, but this works pretty much the same as udev files work.  I might move the udev writing code into a common symbol and use that instead.

Reply:
Oh I remember now - it's because this is libfwupd, so you can't have libfwupdplugin stuff.

Reply:
Ohh, gahh, indeed. It seems like the wrong "layer" to actually set the value; does this need to be in the client lib? It seems to be that only root would be able to write it anyway. a F**u**BiosAttr that subclasses F**wupd**BiosAttr might be a lot cleaner.

Reply:
Ah you know what you're right.  Maybe that's the root of this problem.  It should be the _engine_ writing the file.  Plugin never will and client never will.  It shouldn't be in any lib.

Reply:
I'm fine with it going in libfwupdplugin for namespacing reasons if that helps.